### PR TITLE
Merge form rework

### PR DESCRIPTION
**Pull request by JasonXu314**

### CODE REVIEW

File: pages/application.tsx [hunk 1]
```diff
@@ -83,185 +63,190 @@ const Application: NextPage = () => {
...
+											Next
+										</Button>
+									</Group>
+								</Stack>
+							</Box>
+						</Stepper.Step>
+						<Stepper.Step label="Consent">
+							<Box sx={{ maxWidth: 600 }} mx="auto">
+								<Stack>
+									<Checkbox
```
Comment (Cire15) [inline pages/application.tsx:196]: Although these checkboxes are functionally required, there is no asterisk by their label explicitly indicating them as such before you press submit. They aren't strictly necessary so it's fine without it, but if you can do it than we can put that fix on this pull request.

File: pages/application.tsx [hunk 2]
```diff
@@ -83,185 +63,190 @@ const Application: NextPage = () => {
...
+							<Box sx={{ maxWidth: 600 }} mx="auto">
+								<Stack>
+									<NativeSelect
+										required
+										label="T-Shirt Size"
+										{...form.getInputProps('shirtSize')}
+										data={['XS', 'S', 'M', 'L', 'XL', 'XXL']}
+									/>
+									<MultiSelect
+										label="Dietary Restrictions"
```
Comment (Cire15) [inline pages/application.tsx:125]: Food options should be required, and also I want to try this feature.
```suggestion
required
label="Dietary Restrictions"
```

Comment (Cire15) [reply pages/application.tsx:125]: I don't know how it's gonna format the suggestion, full disclosure

File: pages/application.tsx [hunk 3]
```diff
@@ -83,185 +63,190 @@ const Application: NextPage = () => {
...
-						/>
+						}}>
+						<Stepper.Step label="Personal Info">
+							<Box sx={{ maxWidth: 600 }} mx="auto">
+								<Stack>
+									<TextInput required label="First Name" {...form.getInputProps('firstName')} />
+									<TextInput required label="Last Name" {...form.getInputProps('lastName')} />
+									<TextInput required label="Email" placeholder="your@email.com" {...form.getInputProps('email')} />
+									<TextInput label="Phone Number" {...form.getInputProps('phoneNumber')} />
+									<NativeSelect required label="School" {...form.getInputProps('school')} data={['MST', 'Mizzou', 'Other School']} />
```
Comment (Cire15) [inline pages/application.tsx:86]: We should consider just having this as a TextInput, Other School isn't very descriptive for our statistics. Alternatively, we could have what they had on the quill, which was an extremely large searchable Select with the options stored in a json file. Though, if we were to do that, it's probably large enough for its own branch and pull request.
TLDR: This field is better as an open TextInput than a very limited multiple choice.

Comment (Cire15) [reply pages/application.tsx:86]: Yes I know I was the one who added it in the first place, I see the error of my ways

File: pages/application.tsx [hunk 4]
```diff
@@ -83,185 +63,190 @@ const Application: NextPage = () => {
...
+										}
+										{...form.getInputProps('codeOfConductAgreement')}
+									/>
+									<Checkbox
+										required
+										label={
+											<>
+												I authorize you to share my application/registration information for event administration, ranking, MLH
+												administration, pre- and post-event informational e-mails, and occasional messages about hackathons in-line
+												with the MLH Privacy Policy. I further I agree to the terms of both the
```
Comment (Cire15) [inline pages/application.tsx:214]: The label for this field currently displays like this:
image
The two things to do is make the space before "MLH Contest Terms and Conditions" not part of the hyperlink, and to add a space after the "and the" preceding "MLH Privacy Policy".

**Review by Cire15 — CHANGES_REQUESTED** [review]
The form and the stepper are looking great, it's looking a lot more organized! All the recommended changes are quick and minor, except for the headers of the stepper. I think the grouping of the questions themselves are great, it's just the label names we'll want Logistics on. Fantastic

File: pages/application.tsx [hunk 5]
```diff
@@ -83,185 +63,190 @@ const Application: NextPage = () => {
...
+													setStep(step + 1);
+												}
+											}}>
+											Next
+										</Button>
+									</Group>
+								</Stack>
+							</Box>
+						</Stepper.Step>
+						<Stepper.Step label="Prior Experience">
```
[the remark this answers is not in the document]
Comment (JasonXu314) [reply pages/application.tsx:115]: I put "prior experience" because resume, linkedin, github all kinda seem like they're aimed at gauging the ability of the participant, i'll ask logi what they want to see there.

File: pages/application.tsx (hunk 2 above)
Comment (JasonXu314) [reply pages/application.tsx:125]: i think it would be weird UX if we forced them to pick an option even without dietary restrictions, cuz the vast majority of people won't have dietary restrictions

Comment (Cire15) [reply pages/application.tsx:125]: Yeah, I think so too. Though I think if we want to have it not required as it currently is, we'd have to remove none as an option within dietary restriction. It doesn't really make sense if it's chosen along with the other options, and the absence of a selection means none already.

File: pages/application.tsx (hunk 1 above)
Comment (JasonXu314) [reply pages/application.tsx:198]: there wasnt really a good way to put asterisks built in, so i added spans to the labels, let me know if this is good or i should revert it

File: pages/application.tsx (hunk 3 above)
Comment (JasonXu314) [reply pages/application.tsx:88]: i could make it a text input or i could alternatively add the ability to add schools to the dropdown, which do u think would be better?

File: pages/application.tsx (hunk 1 above)
Comment (Cire15) [reply pages/application.tsx:198]: The spans look good. 👍

File: pages/application.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Cire15) [reply pages/application.tsx:117]: Logi gave the go-ahead on what we already got, so this is a non-issue

**Review by Cire15 — APPROVED** [review]
Ship it, looks good!
